File: src/main/scala/com/qianfeng/stream/Demo01_stream_WordCount.scala
package com.qianfeng.stream

import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
import org.apache.flink.api.scala._

object Demo01_stream_WordCount {
	def main(args: Array[String]): Unit = {
		val env = StreamExecutionEnvironment.getExecutionEnvironment

		val dataStream:DataStream[String] = env.socketTextStream("192.168.5.101", 6666)

		val value = dataStream.flatMap(_.split(" ")).map((_, 1)).keyBy(0).sum(1)


		value.print("wc-")

		env.execute("my_wordcount")

	}
}

File: src/main/java/com/qianfeng/stream/Demo01_stream_wordcount.java
package com.qianfeng.stream;

import org.apache.flink.api.common.functions.FlatMapFunction;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.util.Collector;

public class Demo01_stream_wordcount {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

        DataStreamSource<String> stringDataStreamSource = env.socketTextStream("192.168.5.101", 6666);

        SingleOutputStreamOperator<Tuple2<String, Integer>> sum = stringDataStreamSource.flatMap(new FlatMapFunction<String, Tuple2<String, Integer>>() {

            @Override
            public void flatMap(String s, Collector<Tuple2<String, Integer>> collector) throws Exception {
                String[] s1 = s.split(" ");
                for (String s2 : s1) {
                    collector.collect(new Tuple2<>(s2, 1));
                }
            }
        })
                .keyBy(0)
                .sum(1);

        sum.print("java=wc");
        env.execute("java wordcount");

    }
}
